### PR TITLE
Ffc with adjustments2

### PR DESCRIPTION
**Pull request by ssanderson**

### CODE REVIEW

**Comment by ssanderson** [comment]
@ehebert this is a PR to my own branch of shuffling around the bcolz/sqlite adjustments into reader and writer classes.

File: tests/ffc/test_us_equity_pricing_loader.py [hunk 1]
```diff
@@ -12,287 +12,524 @@
...
+    BcolzDailyBarReader,
+    BcolzDailyBarWriter,
+    SQLiteAdjustmentReader,
+    SQLiteAdjustmentWriter,
 )
+from zipline.finance.trading import TradingEnvironment
+
+UINT_32_MAX = iinfo(uint32).max
+
+# Test calendar ranges over the month of June 2015
```
Comment (ssanderson) [inline tests/ffc/test_us_equity_pricing_loader.py]: @ehebert I expanded out the test range to the whole month of June because some of the tests I wanted to run needed more than 4 days to test all the boundary conditions.

File: tests/ffc/test_us_equity_pricing_loader.py [hunk 2]
```diff
@@ -12,287 +12,524 @@
...
+                    continue
+                data[i, j] = TestingDailyBarWriter.expected_value(
+                    asset,
+                    date,
+                    column,
+                )
+        return data
+
+    def _check_read_results(self, columns, assets, start_date, end_date):
+        result = self.writer.write(self.dest, self.trading_days, self.assets)
```
Comment (ssanderson) [inline tests/ffc/test_us_equity_pricing_loader.py]: `result` is a bad name for this variable; should be `table`

File: tests/ffc/test_us_equity_pricing_loader.py [hunk 3]
```diff
@@ -300,71 +537,32 @@ class UsEquityPricingLoaderTestCase(TestCase):
...
-            np.array([34.004, 34.005, np.nan, np.nan, np.nan]),
-            # Asset 4 should have data starting on the second day and ending
-            # on the third.
-            np.array([np.nan, 44.001, 44.002, np.nan, np.nan]),
-            # Asset 4 should have data starting on the second day through the
-            # end of the range
-            np.array([np.nan, 54.001, 54.002, 54.003, 54.004]),
-            # Asset 5 should have no data
-            np.array([np.nan, np.nan, np.nan, np.nan, np.nan])
+    def setUp(self):
```
Comment (ssanderson) [inline tests/ffc/test_us_equity_pricing_loader.py]: This should just be in `setUpClass`

File: zipline/data/dataset.py [hunk 4]
```diff
@@ -96,6 +96,7 @@ def __new__(mcls, name, bases, dict_):
 
     @property
     def columns(self):
+
```
Comment (ssanderson) [inline zipline/data/dataset.py]: Remove extra whitespace.

File: zipline/data/ffc/loaders/_us_equity_pricing.pyx [hunk 5]
```diff
@@ -12,19 +12,41 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+from cpython cimport (
```
Comment (ssanderson) [inline zipline/data/ffc/loaders/_us_equity_pricing.pyx]: These are unused.  Remove.

File: zipline/data/ffc/loaders/us_equity_pricing.py [hunk 6]
```diff
@@ -25,124 +56,433 @@
...
+            entries in `dates` for which the asset did not yet exist.
+        """
+        query_start = self._calendar.get_loc(dates[0])
+        query_stop = self._calendar.get_loc(dates[-1])
+
+        # Sanity check that the requested date range matches our calendar.
+        # This could be removed in the future if it's materially affecting
+        # performance.
+        query_dates = self._calendar[query_start:query_stop + 1]
+        if not (query_dates.values == dates.values).all():
```
Comment (ssanderson) [inline zipline/data/ffc/loaders/us_equity_pricing.py]: This probably should be a != and an `any()`.

**Comment by ehebert** [comment]
Looks great to me @ssanderson. Glad to have these interfaces more clearly defined.

A few questions/comments here and there, but nothing blocking merging and building on top of this branch.

**Comment by ssanderson** [comment]
@ehebert merged these changes back to my branch, thanks
